### PR TITLE
Fix #10395 Add support of incomplete Social Media URLs 

### LINKED ISSUE #10395: Support account names in contact:* Now it support only full links.
# 🚀 feature request

### Description

https://wiki.openstreetmap.org/wiki/Key:contact , section Social Media / Messengers / Web Services describes value as account name or full URL. Osmand works correctly only if full URL. If account name then osmand only shows it as text but does not create hyperlink to open contact.

### Describe the solution you'd like

Each social network should have predefined url template in code and Osmand should use it if value of tag does not contain **http(s)://**

For example 
contact:instagram - https://www.instagram.com/{value}/
contact:vk - https://vk.com/{value}
contact:facebook - https://www.facebook.com/{value}
contact:twitter - https://twitter.com/{value}
contact:youtube - https://www.youtube.com/c/{value}
contact:ok - https://ok.ru/{value}
contact:telegram - https://tttttt.me/{value}

### PR DESCRIPTION
**Pull request by cepprice**
[Demo](https://youtu.be/UjO2DE9Z3Qo)

### CODE REVIEW

File: OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java [hunk 1]
```diff
@@ -877,12 +896,53 @@ public void onClick(View view) {
 		return new CollapsableView(view, this, collapsed);
 	}
 
+	private String getSocialMediaUrl(String key, String value) {
+		// Regex for url without protocol, with at least 2 valid domains, slash and at least one
+		// char after it (e.g. a-c.com/a)
+		String urlMask = "[a-zA-Z0-9]+[-]*[a-zA-Z0-9]+(\\.[a-zA-Z0-9]+[-]*[a-zA-Z0-9]+)+\\/.+";
+
+		if (Pattern.matches(urlMask, value)) {
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java:904]: is it needed at all? Are there real examples of such broken data?
Looks like unnecessary complication - btw there are ready to use URL parser new URL()

File: OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java [hunk 2]
```diff
@@ -550,16 +560,16 @@ public void buildInternal(View view) {
...
 				row = new AmenityInfoRow(key, R.drawable.ic_action_note_dark, textPrefix,
-						vl, collapsable, collapsableView, 0, false, true,
-						true, 0, "", false, false, matchWidthDivider, 0);
+						vl, null, collapsable, collapsableView, 0, false,
+						true, true, 0, "", false, false, matchWidthDivider, 0);
 			} else if (icon != null) {
-				row = new AmenityInfoRow(key, icon, textPrefix, vl, collapsable, collapsableView,
-						textColor, isWiki, isText, needLinks, poiTypeOrder, poiTypeKeyName,
-						isPhoneNumber, isUrl, matchWidthDivider, 0);
+				row = new AmenityInfoRow(key, icon, textPrefix, vl, socialMediaUrl, collapsable,
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java:566]: is it different code style applied comparing to other team members?

Comment (cepprice) [reply OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java:566]: Do you mean I've changed position (not order) of constructor arguments?

File: OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java (hunk 1 above)
Comment (cepprice) [reply OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java:904]: I think this check is reasonable. From my POV this is the only way to distinguish username (and what looks like it) and url without protocol.

Comment (vshcherb) [reply OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java:904]: user name can't contain '/' as I mentioned previously

File: OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java [hunk 3]
```diff
@@ -877,12 +895,45 @@ public void onClick(View view) {
...
+			sb.deleteCharAt(0);
+		}
+		int lastIdx = sb.length() - 1;
+		if (sb.charAt(lastIdx) == '/') {
+			sb.deleteCharAt(lastIdx);
+		}
+		
+		// It cannot be username
+		if (sb.indexOf("/") != -1) {
+			return "https://" + value;
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java:911]: why do we add https in the beginning?

File: OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java [hunk 4]
```diff
@@ -86,18 +86,20 @@ public AmenityMenuBuilder(@NonNull MapActivity mapActivity, final @NonNull Ameni
 	protected void buildNearestWikiRow(View view) {
 	}
 
-	private void buildRow(View view, int iconId, String text, String textPrefix,
+	private void buildRow(View view, int iconId, String text, String textPrefix, String socialMediaUrl,
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java:89]: why do we need new url ? socialMediaUrl - it looks like it's the same as url

**Pull request merged**